design/schedule-unpaused-triggers-config_design.md

```diff
@@ -0,0 +1,102 @@
+
+## Non Goals
+- Changing the default behavior
+
+
+## High-Level Design
+Add a new field with to the schedule spec to allow user to configure the behavior defaulting to prior behavior of triggering backup immediately.
+
+## Detailed Design
+### CLI Changes
+`velero schedule unpause` will now take an optional bool flag `--unpause-triggers` to allow user to override the behavior configured in the schedule spec.
+
+`velero schedule unpause schedule-1 --unpause-triggers=false` will unpause the schedule but not trigger the backup immediately.
+
+`velero schedule unpause schedule-1 --unpause-triggers=true` will unpause the schedule and trigger the backup immediately.
+
+`velero schedule unpause schedule-1` will check `.spec.UnpauseTriggers` in the schedule to determine behavior. This field will default to true to maintain prior behavior.
+
+### API Changes
+`pkg/apis/velero/v1/schedule_types.go`
+```diff
+// ScheduleSpec defines the specification for a Velero schedule
+type ScheduleSpec struct {
+	// Template is the definition of the Backup to be run
+	// on the provided schedule
+	Template BackupSpec `json:"template"`
+
+	// Schedule is a Cron expression defining when to run
+	// the Backup.
+	Schedule string `json:"schedule"`
+
+	// UseOwnerReferencesBackup specifies whether to use
+	// OwnerReferences on backups created by this Schedule.
+	// +optional
+	// +nullable
+	UseOwnerReferencesInBackup *bool `json:"useOwnerReferencesInBackup,omitempty"`
+
+	// Paused specifies whether the schedule is paused or not
+	// +optional
+	Paused bool `json:"paused,omitempty"`
+
++	// UnpauseTriggers specifies whether to trigger backup immediately when schedule is unpaused
++	// +kubebuilder:default:=true
++	UnpauseTriggers bool `json:"unpauseTriggers,omitempty"`
+}
+```
+
+`LastUnpaused` will be added to `ScheduleStatus` struct to track the last time a schedule was unpaused to support configurable unpause behavior.
+```diff
+// ScheduleStatus captures the current state of a Velero schedule
+type ScheduleStatus struct {
+	// Phase is the current phase of the Schedule
+	// +optional
+	Phase SchedulePhase `json:"phase,omitempty"`
+
+	// LastBackup is the last time a Backup was run for this
+	// Schedule schedule
+	// +optional
+	// +nullable
+	LastBackup *metav1.Time `json:"lastBackup,omitempty"`
+
++	// LastUnpaused is the last time a Schedule was unpaused
++	// +optional
++	// +nullable
++	LastUnpaused *metav1.Time `json:"lastUnpaused,omitempty"`
+
+	// ValidationErrors is a slice of all validation errors (if
+	// applicable)
+	// +optional
+	ValidationErrors []string `json:"validationErrors,omitempty"`
+}
+```
+
+`LastUnpaused` will be set to the current time when a schedule is unpaused. When a schedule is paused, `LastUnpaused` will be set to nil.
+
+## Alternatives Considered
+
+`velero server` flags to configure the behavior. This is not ideal as it is a global setting. This behavior should be configurable by each schedule author.
```

Comment:
Lets also add the pro for the alternative that if its a server setting then the user only has to configure this once.

Reply:
This would be a benefit if there's a single user of this velero server I agree. Could certainly be included as well?

Reply:
yes agreed.

Reply:
Server flag is now included in the design.